client/dhcp_server.go

```diff
@@ -0,0 +1,98 @@
+package client
+
+import (
+	"fmt"
+	"log"
+)
+
+// DhcpServer represents DHCP server resource
+type DhcpServer struct {
+	Id            string `mikrotik:".id"`
+	Name          string `mikrotik:"name"`
+	Disabled      bool   `mikrotik:"disabled"`
+	AddArp        bool   `mikrotik:"add-arp"`
+	AddressPool   string `mikrotik:"address-pool"`
+	Authoritative string `mikrotik:"authoritative"`
+	Interface     string `mikrotik:"interface"`
+	LeaseScript   string `mikrotik:"lease-script"`
+}
+
+func (client Mikrotik) AddDhcpServer(d *DhcpServer) (*DhcpServer, error) {
+	c, err := client.getMikrotikClient()
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := Marshal("/ip/dhcp-server/add", d)
+	log.Printf("[INFO] Running the mikrotik command: `%s`", cmd)
+	r, err := c.RunArgs(cmd)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] command returned: %v", r)
+
+	return client.FindDhcpServer(d.Name)
+}
+
+func (client Mikrotik) UpdateDhcpServer(d *DhcpServer) (*DhcpServer, error) {
+	c, err := client.getMikrotikClient()
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := Marshal("/ip/dhcp-server/set", d)
+	log.Printf("[INFO] Running the mikrotik command: `%s`", cmd)
+	r, err := c.RunArgs(cmd)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] command returned: %v", r)
+
+	return client.FindDhcpServer(d.Name)
+}
+
+func (client Mikrotik) FindDhcpServer(name string) (*DhcpServer, error) {
+	c, err := client.getMikrotikClient()
+
+	if err != nil {
+		return nil, err
+	}
+	cmd := []string{"/ip/dhcp-server/print", "?name=" + name}
+	log.Printf("[INFO] Running the mikrotik command: `%s`", cmd)
+	r, err := c.RunArgs(cmd)
+
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] Found record: %v", r)
+	record := DhcpServer{}
+	err = Unmarshal(*r, &record)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if record.Name == "" {
+		return nil, NewNotFound(fmt.Sprintf("record `%s` not found", name))
```

Comment:
```suggestion
		return nil, NewNotFound(fmt.Sprintf("dhcp server `%s` not found", name))
```

Minor suggestion to make sure this error message is obvious it applies to the DHCP server.

Reply:
thanks for suggestion.
I was thinking in terms of `resourceWrapper` where resource type is not printed in errors (probably, I will change this behavior once that PR is ready for review)